armory/data/utils.py

```diff
@@ -81,6 +84,29 @@ def maybe_download_weights_from_s3(weights_file: str) -> str:
                         "`saved_model_dir` directory on your host environment."
                     )
                 )
+
+    if auto_expand_tars:
+        if tarfile.is_tarfile(filepath):
+            logger.debug(f"Detected model weights file {weights_file} as a tar archive")
+            with tarfile.open(filepath) as tar:
+                # check if the tarfile contains a directory containing all its members
+                # ie if the tarfile expands out entirely into a subdirectory
+                dirs = [fi.name for fi in tar.getmembers() if fi.isdir()]
+                for d in dirs:
+                    if all(name.startswith(d) for name in tar.getnames()):
```

Comment:
If two subdirs share a common prefix, like "test" and "test2", this will return "test", even though there are two subdirectories. This is because `startswith` is just a string comparison. I propose doing this with `os.path`:

```
dirs = [fi.name for fi in tar.getmembers() if fi.isdir()]
commonpath = os.path.commonpath(tar.getnames())
if not commonpath or commonpath not in dirs:
    raise PermissionError(...)

```